### PR TITLE
Slightly refactor for intellij-aya

### PR DESCRIPTION
**Pull request by imkiva**
bors merge

### CODE REVIEW

File: lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.Serializable;
+
+/**
+ * Special gift for Intellij IDEA plugin. Usually used in response models to skip json serialization of certain data.
+ * @see org.aya.lsp.models.HighlightResult.Symbol
+ */
+public final class ForIDEA<T> implements Serializable {
```
Comment (ice1000) [inline lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java:13]: Isn't this `Ref<T>`?

**Review by ice1000 — COMMENTED** [review]
To Kiva's question

File: lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java (hunk 1 above)
Comment (imkiva) [reply lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java:13]: The inner instance of `Ref<T>` is not transient. And the purpose of `ForIDEA` is skipping serialization when serving VSCode

Comment (ice1000) [reply lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java:13]: I would suggest asking kala-common to implement a transient Ref, what do you think

Comment (imkiva) [reply lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java:13]: I agree with you. /cc @Glavo

**Pull request merged**